config/initializers/rack_attack.rb

```diff
@@ -71,4 +71,32 @@ def self.auth_token(request)
   path = payload.fetch(:request).fullpath
 
   Rails.logger.warn("[rack-attack] Throttled request #{request_id} from #{ip} to '#{path}'")
+
+  # Web requests are sent to BigQuery via a concern in the ApplicationController.
+  # If Rack intercepts the request it won't reach the controller, so we need
+  # to manually send web requests that are rate limited to BigQuery.
+  if DfE::Analytics.enabled?
+    request = ActionDispatch::Request.new(payload[:request].env)
+    response = ActionDispatch::Response.new(429)
+
+    # Fetch API user
+    token = request.authorization.to_s.split("Bearer ").last
```

Comment:
I think we mentioned maybe having a service where we calculate logic for events so we don't have this dotted in many places?